nav2_behavior_tree/plugins/action/remove_passed_goals_with_multiple_planners_action.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2021 Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <limits>
+
+#include "nav_msgs/msg/path.hpp"
+#include "nav2_util/geometry_utils.hpp"
+
+#include "nav2_behavior_tree/plugins/action/remove_passed_goals_with_multiple_planners_action.hpp"
+
+namespace nav2_behavior_tree
+{
+
+RemovePassedGoalsWithMultiplePlanners::RemovePassedGoalsWithMultiplePlanners(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::ActionNodeBase(name, conf),
+  viapoint_achieved_radius_(0.5)
+{
+  getInput("radius", viapoint_achieved_radius_);
+
+  getInput("global_frame", global_frame_);
+  getInput("robot_base_frame", robot_base_frame_);
+  
+  tf_ = config().blackboard->get<std::shared_ptr<tf2_ros::Buffer>>("tf_buffer");
+  auto node = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+  node->get_parameter("transform_tolerance", transform_tolerance_);
+}
+
+inline BT::NodeStatus RemovePassedGoalsWithMultiplePlanners::tick()
+{
+  setStatus(BT::NodeStatus::RUNNING);
+
+  auto prev_planner_ids = planner_ids_;
+  getInput("input_planner_ids", planner_ids_);
+
+  // If input planners changed(e.g by another BT node), then we should populate remaining list of planners again.
+  if (prev_planner_ids != planner_ids_){
+    remaining_planner_ids_ = planner_ids_;
+  }  
+
+  
+  Goals goal_poses;
+  getInput("input_goals", goal_poses);
+
+  if (goal_poses.empty()) {
+    setOutput("output_goals", goal_poses);
+    return BT::NodeStatus::SUCCESS;
+  }
+  if (remaining_planner_ids_.size() != goal_poses.size() && planner_ids_.size()  != 1 && planner_ids_.size() != 0){
+    RCLCPP_WARN(
```

Comment:
This should throw an exception, that's a critical run time failure condition. Failing a BT node has consequences of a recoverable action through the tick logic that this couldn't do. The navigation request should be halted